csharp/src/Google.Protobuf.Test/FieldMaskTreeTest.cs

```diff
@@ -421,7 +422,7 @@ public void Merge(bool useDynamicMessage)
             destination = source.Clone();
             Merge(new FieldMaskTree().AddFieldPath("payload.single_int32"),
                 sourceWithPayloadInt32Unset, destination, options, useDynamicMessage);
-            Assert.AreEqual(0, destination.Payload.SingleInt32);
+            Assert.AreEqual(1234, destination.Payload.SingleInt32);
```

Comment:
According to the docs, this should not have been overwritten because a default value should not be overwritten if `ReplacePrimitiveFields == false`.

Reply:
I disagree: it's in the field mask, which means it comes under the bit of the docs saying:

> Updating to set a default value in this case requires some external mechanism, such as FieldMask.

Basically I interpret the field mask as saying "Please make sure that SingleInt32 is copied from the source, even if it wasn't set there."

Reply:
My interpretation was adding more context to the docs to say something more like:
> Updating to set a default value in this case requires some external mechanism, such as FieldMask _with ReplacePrimitiveFields set to true (italics mine)_.

The docs still are accurate, they're just not as specific as they could be.

Reply:
FieldMask doesn't have the concept of ReplacePrimitiveFields though - only FieldMaskTree does. A field mask is just a list of field paths - it doesn't have any options.